chanzuckerberg/czgenepi
# Thuang 132238 virtual scrolling

RDEV: https://thuang-virtual-scroll-frontend.dev.genepi.czi.technology/data/samples

### Description

1. Rewrite `DataTable` in order to allow virtual scrolling (turned out that `<table />` related elements don't work for virtual scrolling). Current implementation is hand rolled, but eventually we will use Material UI's table if needed
3. Add `react-window` for virtual scrolling
4. Some visual fixes were pulled in from "Visual Fixes" branch to make styling the table easier, such as:
    1. Updating all the theme related variables to match IDseq design system's values (space, typography, corner, etc.)
    2. Remove global style that makes every `div` a flex-box, since it makes styling very difficult and scatters style definitions across multiple places

#### Issue
https://app.clubhouse.io/genepi/story/132238/look-into-rendering-less-items-for-sample-table-view-pagination-or-another-faster-load-fix

### Test plan

![demo](https://user-images.githubusercontent.com/6309723/114918354-d745b300-9ddb-11eb-907a-4f49355d93c2.gif)


1. Log in
2. Load Samples table, you should now see the new virtual scrolling table
2. Switching between Samples and Trees should see no freezing anymore
3. Search/filter should not freeze anymore too
4. UPDATE: Please check the load state too!

![demo](https://user-images.githubusercontent.com/6309723/114963309-501b2e00-9e21-11eb-82ce-1bf0795d95fe.gif)

PTAL thank you 🎉 !
